### PR TITLE
- Added __getitem__ and __len__ to the Cigar class, pulling the answer…

### LINKED ISSUE #153: Feature request: add __get_item__() to Cigar class
This will enable users to access the i-th element of `cigar` with `cigar[i]` rather than `cigar.elements[i]`

### PR DESCRIPTION
**Pull request by yfarjoun**
… from the .elements field

- Added tests and documentation
- fixes #153

### CODE REVIEW

File: fgpyo/sam/__init__.py [hunk 1]
```diff
@@ -111,6 +111,25 @@
...
+    >>> tuple(x.operator.character for x in cigar[1:3])
+    ('D','M')
+    >>> tuple(x.operator.character for x in cigar[-2:])
+    ('I', 'S')
+
+
+The Cigar has __len__ defined to be the length of the elements:
+    >>> cigar = Cigar.from_cigarstring("50M2D5M2I10S")
+    >>> len(cigar) == len(cigar.elements)
+    True
```
Comment (tfenne) [inline fgpyo/sam/__init__.py:132]: I actually would prefer we don't define __len__.  I think it's likely to lead to confusion since we have so many different lengths in question:
- `len(cigar.elements)`
- `sum(c.length for c in cigar.elements)`
- `cigar.length_on_target()`
- `cigar.length_on_query()`

File: fgpyo/sam/__init__.py [hunk 2]
```diff
@@ -543,6 +562,27 @@ def length_on_target(self) -> int:
         """Returns the length of the alignment on the target sequence."""
         return sum([elem.length_on_target for elem in self.elements])
 
+    def __getitem__(
+        self, index: Union[int, slice]
+    ) -> Union[CigarElement, Tuple[CigarElement, ...]]:
+        """Returns the cigar element indexed by index
```
Comment (tfenne) [inline fgpyo/sam/__init__.py:568]:
```suggestion
        """Returns the cigar element(s) selected by index
```

File: fgpyo/sam/__init__.py [hunk 3]
```diff
@@ -543,6 +562,27 @@ def length_on_target(self) -> int:
...
+
+        Arguments:
+            index: int The index of the requested cigar-element
```
Comment (tfenne) [inline fgpyo/sam/__init__.py:571]:
```suggestion
            index: int The index of the requested cigar element(s)
```

File: fgpyo/sam/__init__.py [hunk 4]
```diff
@@ -543,6 +562,27 @@ def length_on_target(self) -> int:
...
+        Throws:
+            TypeError if index isn't an integer or a slice
+            IndexError: if there's no such element
+
+        """
+        return self.elements[index]
+
+    def __len__(self) -> int:
+        """Returns the number of elements in the cigar"""
+        return len(self.elements)
```
Comment (tfenne) [inline fgpyo/sam/__init__.py:584]: Sorry, but do not want.

File: fgpyo/sam/tests/test_cigar.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+cigar = Cigar.from_cigarstring("1M4D45N37X23I11=")
+
+
+def test_cigar_length_exists() -> None:
+    assert len(cigar) == len(cigar.elements)
+
+
+@pytest.mark.parametrize("index", range(-len(cigar), len(cigar)))
+def test_direct_access(index: int) -> None:
```
Comment (tfenne) [inline fgpyo/sam/tests/test_cigar.py:13]: Could you please add a simple test that indexing out of range fails as expected?

**Review by tfenne — CHANGES_REQUESTED** [review]

**Review by tfenne — APPROVED** [review]
